### PR TITLE
Added Russian snils generator

### PR DESCRIPTION
**Pull request by eter4eter**
Added Russian snils generator which returns valid values.

### CODE REVIEW

**Comment by lk-geimfari** [comment]
@eter4eter Thanks for PR. Please, check [details](https://travis-ci.org/lk-geimfari/mimesis/builds/387268014?utm_source=github_status&utm_medium=notification) to understand why tests failed.

Seems like you forgotten to update tests and correctly sort imports.

**Comment by lk-geimfari** [comment]
@eter4eter Yeah, that's right. Thanks!

File: mimesis/builtins/ru.py [hunk 1]
```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
...
-
         :return: SNILS.
 
         :Example:
             451-952-540-41.
         """
-        mask = '###-###-###-##'
-        return self.random.custom_code(mask=mask)
+        snils = [self.random.randint(0, 9) for _ in range(9)]
+        while int(reduce(lambda x, y: x + y,
```
Comment (sobolevn) [inline mimesis/builtins/ru.py]: Sorry, but this code looks to complicated for me.
Could we change how this loop works?

Things that are not clear for me:
1. what is `1001998`? if it is a constant let's give it a name
2. why do we have `snils = [self.random.randint(0, 9) for _ in range(9)]` twice?
3. can we please not use `reduce` and `map` with `lambda`s? In my personal opinion their usage is almost never required in `python`

File: mimesis/builtins/ru.py [hunk 2]
```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
...
+                         map(lambda x: str(x), snils))) <= 1001998:
+            snils = [self.random.randint(0, 9) for _ in range(9)]
+
+        def get_snils_sum(snils):
+            s = 0
```
Comment (sobolevn) [inline mimesis/builtins/ru.py]: What is `s`? Can we please give this variable a more meaningful name?

File: mimesis/builtins/ru.py [hunk 3]
```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
...
+            for index, mult in enumerate(range(9, 0, -1)):
+                s += snils[index] * mult
+
+            def ret_s(s):
+                if s < 100:
+                    if s < 10:
+                        return '0' + str(s)
+                    return str(s)
+                elif 100 <= s <= 101:
+                    return '00'
```
Comment (sobolevn) [inline mimesis/builtins/ru.py]: Can you please explain what happens here?

File: mimesis/builtins/ru.py [hunk 4]
```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
...
+        while int(reduce(lambda x, y: x + y,
+                         map(lambda x: str(x), snils))) <= 1001998:
+            snils = [self.random.randint(0, 9) for _ in range(9)]
+
+        def get_snils_sum(snils):
+            s = 0
+            for index, mult in enumerate(range(9, 0, -1)):
+                s += snils[index] * mult
+
+            def ret_s(s):
```
Comment (sobolevn) [inline mimesis/builtins/ru.py]: There are several things with this function that could be improved:
1. `ret_s` is not a very descriptive name, it should be enhanced 
2. `ret_s` has a nested level of 3. That's way to much!

File: mimesis/builtins/ru.py [hunk 5]
```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
...
+                    if s < 10:
+                        return '0' + str(s)
+                    return str(s)
+                elif 100 <= s <= 101:
+                    return '00'
+                else:
+                    return ret_s(s % 101)
+
+            return ret_s(s)
+        snils = reduce(lambda x, y: x + y, map(lambda x: str(x),
```
Comment (sobolevn) [inline mimesis/builtins/ru.py]: The same about `reduce` and `map` goes for this line.

**Comment by sobolevn** [comment]
@eter4eter one more thing. I can see that now we have many path to go inside this function.
But we have not provided any new tests for it.

We need to ensure that coverage would not decrease.

**Comment by sobolevn** [comment]
@lk-geimfari what's wrong with `appveyor`? I could not understand what's happening, since I can not windows.

**Comment by lk-geimfari** [comment]
@sobolevn The same thing. My OS is an Ubuntu, soi don't understand why tests are failed on Windows.

@duckyou There are suggestions why?

**Comment by lk-geimfari** [comment]
@eter4eter Please, check out comments and recreate PR if you still want to contribute.

**Pull request closed without merging**